base/abstractdict.jl

```diff
@@ -563,7 +563,7 @@ end
 empty(d::IdDict, ::Type{K}, ::Type{V}) where {K, V} = IdDict{K,V}()
 
 function rehash!(d::IdDict, newsz = length(d.ht))
-    d.ht = ccall(:jl_idtable_rehash, Any, (Any, Csize_t), d.ht, newsz)
+    d.ht = ccall(:jl_idtable_rehash, Vector{Any}, (Vector{Any}, Csize_t), d.ht, newsz)
```

Comment:
We still use `Any` for the argument type in these cases; there should be no performance difference. That clarifies that we want to pass a julia object reference.

Reply:
> there should be no performance difference

presumably the return type specification would let inference make use of that. Though perhaps that would be better achieved by a type assert.

Reply:
No, I mean there's no performance difference for using `Any` as the *argument* type.

Reply:
OK, but there is a performance difference. Is the idea to fix it elsewhere?

Reply:
Oh, OK (I may have failed to refresh the browser). You're saying you'd support changing the return type declaration but not the argument type?

Reply:
Yes.